src/plugins/ui_actions/public/plugin.ts

```diff
@@ -17,43 +17,30 @@
  * under the License.
  */
 
-import { CoreStart, PluginInitializerContext, CoreSetup, Plugin } from 'src/core/public';
-import { UiActionsApi, ActionRegistry, TriggerRegistry } from './types';
-import { createApi } from './api';
-
-export interface UiActionsSetup {
-  attachAction: UiActionsApi['attachAction'];
-  detachAction: UiActionsApi['detachAction'];
-  registerAction: UiActionsApi['registerAction'];
-  registerTrigger: UiActionsApi['registerTrigger'];
-}
+import { CoreStart, CoreSetup, Plugin, PluginInitializerContext } from 'src/core/public';
+import { UiActionsService } from './service';
+
+export type UiActionsSetup = Pick<
+  UiActionsService,
+  'attachAction' | 'detachAction' | 'registerAction' | 'registerTrigger'
+>;
 
-export type UiActionsStart = UiActionsApi;
+export type UiActionsStart = PublicMethodsOf<UiActionsService>;
 
 export class UiActionsPlugin implements Plugin<UiActionsSetup, UiActionsStart> {
-  private readonly triggers: TriggerRegistry = new Map();
-  private readonly actions: ActionRegistry = new Map();
-  private api!: UiActionsApi;
+  private readonly service = new UiActionsService();
 
-  constructor(initializerContext: PluginInitializerContext) {
-    this.api = createApi({ triggers: this.triggers, actions: this.actions }).api;
-  }
+  constructor(initializerContext: PluginInitializerContext) {}
 
   public setup(core: CoreSetup): UiActionsSetup {
-    return {
-      registerTrigger: this.api.registerTrigger,
-      registerAction: this.api.registerAction,
-      attachAction: this.api.attachAction,
-      detachAction: this.api.detachAction,
-    };
+    return this.service;
```

Comment:
Interesting... I think the public contract is what we _actually_ return, not the _type_ we say we return?

So would you consider it a breaking change to remove something returned here if it wasn't specified in `UiActionsSetup`?  I think perhaps we should only return the bare minimum?

Reply:
I think the contract is what is specified in the TypeScript definition as we are using TypeScript. At least that is how we have it everywhere, otherwise it is almost always possible to get access to more things than what is specified in the TypeScript type, for example, say you return an `Observable` there are plenty of "hidden" methods that are not part of the contract, or even any object in JavaScript has properties like `.hasOwnProperty` and `.toString()`.